nova_ecs/utils_test.ts

```diff
@@ -28,6 +29,43 @@ describe('utils', () => {
             }
         });
 
+            
+        const elems:[string,Set<string>][] = [['a',new Set()],
+                                              ['b',new Set(['a'])],
+                                              ['c',new Set(['b'])],
+                                              ['d',new Set(['c'])]];
+        function* permutations(n:number):IterableIterator<number[]>{
+            if (n == 1){
+                yield [0];
+                return;
+            }
+            for (let i = 0; i < n; i++){
+                for (let p of permutations(n-1)){
+                    yield [i,... p.map(v => (v>=i?v+1:v))];
```

Comment:
This is cool, but please add a comment explaining how it works. Just something like "Choose where `i` goes, and then recurse to choose the positions of the rest of the numbers. Add 1 to make the rest of the numbers never overlap with `i`."